include/nbl/builtin/hlsl/bxdf/common.hlsl

```diff
@@ -48,358 +94,883 @@ struct Basic
 namespace surface_interactions
 {
 
-template<class RayDirInfo>
-struct Isotropic
+#define NBL_CONCEPT_NAME Isotropic
+#define NBL_CONCEPT_TPLT_PRM_KINDS (typename)(typename)(typename)
+#define NBL_CONCEPT_TPLT_PRM_NAMES (T)(B)(U)                        // B is type Basic<T>
```

Comment:
if you require `Basic<B>` to have a `vector_type` and `scalar_type` you don't need the B and U parameters